src/main/java/vendingmachine/model/HoldingCoins.java

```diff
@@ -0,0 +1,91 @@
+    private final Map<Coin, Integer> coins;
+
+    public HoldingCoins(String inputMoney) {
+        isValidMoney(inputMoney);
+        coins = randomPickCoinsInList(inputMoney);
+    }
+
+    private void isValidMoney(String inputMoney) {
+        isNumber(inputMoney);
+        isPositiveNumber(inputMoney);
+    }
+
+    private void isNumber(String inputMoney) {
+        if (!inputMoney.chars().allMatch(Character::isDigit)) {
+            throw new IllegalArgumentException("[ERROR] 보유 금액은 숫자여야 합니다.");
+        }
+    }
+
+    private void isPositiveNumber(String inputMoney) {
+        if (Integer.parseInt(inputMoney) < 0) {
+            throw new IllegalArgumentException("[ERROR] 보유 금액은 양수여야 합니다.");
+        }
+    }
+
+    private Map<Coin, Integer> randomPickCoinsInList(String inputMoney) {
+        Map<Coin, Integer> randomCoinHashMap = new TreeMap<>();
+        int currentMoney = Integer.parseInt(inputMoney);
+        while (currentMoney >= usableCoinList.stream()
+                .min(Comparator.comparing(coin -> coin))
+                .orElseThrow(NoSuchElementException::new)) {
+            int pickedCoinValue = Randoms.pickNumberInList(usableCoinList);
+            if (currentMoney >= pickedCoinValue) {
+                currentMoney -= pickedCoinValue;
+                Coin key = getCoinByAmount(pickedCoinValue);
+                randomCoinHashMap.put(key, randomCoinHashMap.getOrDefault(key, 0) + 1);
+            }
+        }
+        return randomCoinHashMap;
+    }
+
+    private Coin getCoinByAmount(int amount) {
+        if (amount == Coin.COIN_500.getAmount()) {
+            return Coin.COIN_500;
+        }
+        if (amount == Coin.COIN_100.getAmount()) {
+            return Coin.COIN_100;
+        }
+        if (amount == Coin.COIN_50.getAmount()) {
+            return Coin.COIN_50;
+        }
+        return Coin.COIN_10;
+    }
+
+    public int getCoinCount(Coin coin) {
+        return this.coins.getOrDefault(coin, 0);
+    }
+
+    public Map<Coin, Integer> getCoins() {
+        return new HashMap<>(this.coins);
+    }
+
+    public Map<Coin, Integer> getChanges(InsertedMoney insertedMoney) {
+        Map<Coin, Integer> changes = new TreeMap<>();
+        for (Map.Entry<Coin, Integer> coinAndCount : coins.entrySet()) {
+            Coin coin = coinAndCount.getKey();
+            int count = coinAndCount.getValue();
+            int possibleCoinCount = getPossibleCoinCount(insertedMoney.getInsertedMoney(), coin.getAmount(), count);
+            insertedMoney.calculateChange(possibleCoinCount * coin.getAmount());
+            if(possibleCoinCount > 0)
+                changes.put(coin, possibleCoinCount);
+        }
+        return changes;
+    }
+
+    public int getPossibleCoinCount(int insertedMoney, int amount, int count) {
+        int returnCount = 0;
+        while (insertedMoney >= amount) {
+            insertedMoney -= amount;
+            returnCount += 1;
```

Comment:
```suggestion
            returnCount++;
```
이렇게 하는게 좋지 않을까요?
추가로 return 이라는 네이밍은 좋아보이지 않네요! 😣

Reply:
앗, 넵!
처음에 ++했다가, ide에서 빨간 줄이 떴어서 바꿨던 기억이 있습니다 ㅜㅜ 시간이 없었어서 그냥 += 1로 바꾸고 넘어갔습니다.

네이밍을 좀 더 신경쓰겠습니다..! 확실히 전체적으로 서둘러서 네이밍 생각나는대로 막 코드를 작성했더니 클린하지 못한 것 같습니다 ㅜㅜ
추가적으로 역시 이 메서드에서도 get이라는 접두어를 쓰면 안되겠네요!